dlib/rand/rand_kernel_1.h

```diff
@@ -265,6 +265,31 @@ namespace dlib
                 return r.real();
             }
 
+            double get_random_exponential (
+                double lambda
+            )
+            {
+                DLIB_ASSERT(lambda > 0, "lambda must be greater than zero");
+                double u = 0.0;
+                while (u == 0.0)
+                    u = get_random_double();
+                return -std::log( u ) / lambda;
+            }
+
+            double get_random_weibull (
+                double alpha,
+                double beta,
+                double gamma
+            )
+            {
+                DLIB_ASSERT(alpha > 0, "alpha must be greater than zero");
+                DLIB_ASSERT(beta > 0, "beta must be greater than zero");
+                double u = 0.0;
+                while (u == 0.0)
+                    u = get_random_double();
+                return gamma + beta*std::pow(-std::log(u), 1.0 / alpha);
+            }
+            
```

Comment:
Please update the docs in the _abstract.h file too.

Reply:
It's also a good idea to make the parameter names match the names used on wikipedia, since that's what most people will expect.  E.g. https://en.wikipedia.org/wiki/Weibull_distribution

Reply:
Shall i refactor alpha and beta to k and lambda respectively?

Reply:
Yeah, seems like a good idea :)